src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -366,6 +366,21 @@ private void CompileMethodInternal(IMethodNode methodCodeNodeNeedingCode, Method
 #endif
             }
 
+            if (codeSize != _code.Length)
+            {
+                // If the generated code is smaller than allocatedHotCodeSize, then trim the codeBlock.
+                //
+                // Currently, hot/cold splitting is not done and hence `codeSize` just includes the size of
+                // hotCode. Once hot/cold splitting is done, need to trim respective `_code` or `_coldCode`
+                // accordingly.
+                if (_compilation.Logger.IsVerbose)
+                {
+                    Log.WriteLine($"Trimming codeSize from {_code.Length} to {codeSize}, Savings: {_code.Length - codeSize}, Method: {_methodCodeNode.Method.Name}");
+                }
+                Debug.Assert(codeSize != 0);
+                Debug.Assert(_code.Length > codeSize);
+                Array.Resize(ref _code, (int)codeSize);
```

Comment:
wonder if there are any page alignment considerations here?

Reply:
Not really. Alignments are taken into account in `PublishCode()`.

https://github.com/dotnet/runtime/blob/e7600e00800e0ee9826ba97146f1da8a1250ccae/src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs#L378-L380